src/plugins/elasticsearch/index.js

```diff
@@ -49,12 +52,28 @@ module.exports = function (kibana) {
         return reply.continue();
       }
 
+      function noCreateIndex(request, reply) {
+        const requestPath = _.trim(request.path);
+        const matchPath = createProxy.createPath(config.get('kibana.index'));
+
+        if (requestPath === matchPath) {
+          return reply(Boom.methodNotAllowed('You cannot modify the primary kibana index through this interface.'));
+        }
+
+        reply.continue();
+      }
+
+      // These routes are actually used to deal with things such as managing
+      // index patterns and advanced settings, but since hapi treats route
+      // wildcards as zero-or-more, the routes also match the kibana index
+      // itself. The client-side kibana code does not deal with creating nor
+      // destroying the kibana index, so we limit that ability here.
```

Comment:
Interesting, I had no idea elasticsearch would create an index automatically in that situation.

In any case, I don't think there's anything we can do about that on Kibana's end.  This change isn't really about security necessarily, so much as making sure Kibana restricts functionality that it doesn't itself use and that the default shield permissions don't allow.